src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -3012,24 +2984,61 @@ void G1CollectedHeap::do_collection_pause_at_safepoint_helper(double target_paus
   // We also do not allow mixed GCs during marking.
   assert(!collector_state()->mark_or_rebuild_in_progress() || collector_state()->in_young_only_phase(), "sanity");
 
+  return collector_state()->in_concurrent_start_gc();
+}
+
+void G1CollectedHeap::set_default_active_worker_threads(){
```

Comment:
The name, `set_default_active_worker_threads`, is quite generic, suggesting it's called from multiple places. However, it's actually only for the incremental evacuation pause, right? If so, it's probably clearer to include "evacuation" in the name, IMO.

Reply:
Renamed to `set_young_collection_default_active_worker_threads`.